src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -164,61 +571,55 @@ public class BDI_MigrationMappingUtility {
         Boolean badIsDeletedField = (existingFieldMapping.Is_Deleted__c != false);
-        Boolean foundMismatch = badMappingSetId || 
-                badRequiredField || 
-                badSourceFieldAPIName || 
-                badTargetFieldAPIName || 
-                badTargetObjectMappingId || 
-                badIsDeletedField;
+        Boolean badMappingSetId = (
+            existingFieldMapping.Data_Import_Field_Mapping_Set__c != DEFAULT_FIELD_MAPPING_SET.Id
+        );
+        Boolean badSourceFieldAPIName = (
+            existingFieldMapping.Source_Field_API_Name__c != newFieldMapping.sourceFieldAPIName
+        );
+        Boolean badTargetFieldAPIName = (
+            existingFieldMapping.Target_Field_API_Name__c != newFieldMapping.targetFieldAPIName
+        );
+        Boolean badTargetObjectMappingId = (
+            existingFieldMapping.Target_Object_Mapping__c != existingDefaultObjectMapping.Id
+        );
+        Boolean foundMismatch = badMappingSetId
+            || badRequiredField
+            || badSourceFieldAPIName
+            || badTargetFieldAPIName
+            || badTargetObjectMappingId
+            || badIsDeletedField;
 
         return foundMismatch;
     }
 
-    /** @description Name of default packaged Object Mapping Set */
-    private static final String DEFAULT_OBJECT_MAPPING_SET_RECORD_NAME = 'Default_Object_Mapping_Set';
-
-    /** @description Name of default packaged Field Mapping Set */
-    private static final String DEFAULT_FIELD_MAPPING_SET_RECORD_NAME = 'Default_Field_Mapping_Set';
-
-    /** @description Default packaged Object Mapping Set metadata record */
-    private static final Data_Import_Object_Mapping_Set__mdt DEFAULT_OBJECT_MAPPING_SET = [SELECT 
-                Id, 
-                DeveloperName, 
-                Label
-        FROM Data_Import_Object_Mapping_Set__mdt
-        WHERE DeveloperName = :DEFAULT_OBJECT_MAPPING_SET_RECORD_NAME];
-
-    /** @description Default packaged Field Mapping Set metadata record */
-    private static final Data_Import_Field_Mapping_Set__mdt DEFAULT_FIELD_MAPPING_SET = [SELECT 
-                Id, 
-                DeveloperName, 
-                Label
-        FROM Data_Import_Field_Mapping_Set__mdt
-        WHERE DeveloperName = :DEFAULT_FIELD_MAPPING_SET_RECORD_NAME];
-
     /**
-     * @return A 10 character unique 'jobId' to use for
+     * @description Check the status of custom metadata deployment
+     * @return Metadata.DeployResult JSON
      */
-    private static String getUniqueJobId() {
-        return UserInfo.getUserId().left(15).right(4).toUpperCase() +
-                String.valueOf(Integer.valueOf(math.rint(math.random()*5000000))).right(6);
+    public String checkDeployStatus(String deploymentId) {
+        Http h = new Http();
+        HttpRequest request = new HttpRequest();
+        request.setEndpoint(Url.getOrgDomainUrl().toExternalForm() +
+            '/services/data/v44.0/metadata/deployRequest/' +
```

Comment:
Hey @ptewson-sfdo - Is it alright to hardcode the api version in this endpoint?

Reply:
I've made my peace with it! Although, maybe we can define the string as a class variable?